### PR TITLE
New Metrics API

### PR DESCRIPTION
**Pull request by sebtomba**
## Overview
* Add Kamon InfluxDB reporter
* Add command-line flags and configuration options for Prometheus and InfluxDB reporters
* Change `Metrics`-API to take an implicit `MetricsSource` parameter with a module name
* Use node-id as metrics hierarchy root name
* Review existing metrics if they make sense and are required
* Add new metrics if required
* Use Kamon System Metrics instead of our own JVM metrics
* Use histograms instead of gauges
* Implement a timer in `Metrics`

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/CORE-1489

### Complete this checklist before you submit the PR
- [ ] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] If this PR adds a new feature, this PR includes tests related to this feature.
- [x] You assigned one person to review this PR

### CODE REVIEW

**Comment by adaszko** [comment]
bors try

**Comment by adaszko** [comment]
bors try

**Comment by sebtomba** [comment]
bors try

File: block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala [hunk 1]
```diff
@@ -17,9 +18,12 @@ class InMemBlockStore[F[_]] private ()(
     metricsF: Metrics[F]
 ) extends BlockStore[F] {
 
+  private implicit val metricsSource: MetricsSource =
+    MetricsSource(RSpaceMetricsSource, "block-storage.in-mem")
```
Comment (lukasz-golebiewski) [inline block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala]: Why the change of prefix from "block-store"?

File: block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala [hunk 2]
```diff
@@ -4,10 +4,11 @@ import cats._
 import cats.effect.Sync
 import cats.effect.concurrent.Ref
 import cats.implicits._
+
 import coop.rchain.blockstorage.BlockStore.BlockHash
 import coop.rchain.casper.protocol.BlockMessage
-import coop.rchain.metrics.Metrics
-
+import coop.rchain.metrics.{Metrics, MetricsSource}
+import coop.rchain.rspace._
```
Comment (lukasz-golebiewski) [inline block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala]: Why is this dependency needed?

File: block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala (hunk 1 above)
Comment (lukasz-golebiewski) [reply block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala]: I don't think `RSpaceMetricsSource` should be used here

File: node/src/main/scala/coop/rchain/node/NodeRuntime.scala [hunk 3]
```diff
@@ -185,17 +187,6 @@ class NodeRuntime private[node] (
...
     } yield ()).unsafeRunSync(scheduler)
 
-  def startReportJvmMetrics(
-      implicit metrics: Metrics[Task],
-      jvmMetrics: JvmMetrics[Task]
-  ): Task[Unit] =
-    Task.delay {
-      import scala.concurrent.duration._
-      loopScheduler.scheduleAtFixedRate(3.seconds, 3.second)(
-        JvmMetrics.report[Task].unsafeRunSync(scheduler)
```
Comment (rabbitonweb) [inline node/src/main/scala/coop/rchain/node/NodeRuntime.scala]: consider removing this in next PR, we no longer need JvmMetrics nor the `"diagnostics"` functionality

File: block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala (hunk 1 above)
Comment (rabbitonweb) [reply block-storage/src/main/scala/coop/rchain/blockstorage/InMemBlockStore.scala]: this is `storage` module this different source should be used

File: node/src/test/scala/coop/rchain/node/configuration/toml/TomlConfigurationSpec.scala [hunk 4]
```diff
@@ -48,6 +48,16 @@ class TomlConfigurationSpec extends FunSuite with Matchers {
       |approve-genesis-duration = "30min"
       |approve-genesis-interval = "1min"
       |deploy-timestamp = 1
+      |
+      |[kamon]
+      |prometheus = false
+      |influx-db = true
```
Comment (rabbitonweb) [inline node/src/test/scala/coop/rchain/node/configuration/toml/TomlConfigurationSpec.scala]: Consider renaming to `influx-db-enabled`
